test/widgets/message_list_test.dart

```diff
@@ -922,6 +922,72 @@ void main() {
         await tester.pump();
         tester.widget(find.text('new stream name'));
       });
+
+      testWidgets('navigates to TopicNarrow on tapping topic in CombinedFeedNarrow', (tester) async {
+        final pushedRoutes = <Route<void>>[];
+        final navObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+        final channel = eg.stream();
+        final message = eg.streamMessage(stream: channel, topic: 'topic name');
+
+        await setupMessageListPage(
+          tester,
+          narrow: const CombinedFeedNarrow(),
+          messages: [message],
+          subscriptions: [eg.subscription(channel)],
+          navObservers: [navObserver],
+        );
+
+        assert(pushedRoutes.length == 1);
+        pushedRoutes.clear();
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          messages: [message],
+          foundOldest: true,
+        ).toJson());
+
+        final topicFinder = find.descendant(
+          of: find.byType(StreamMessageRecipientHeader),
+          matching: find.text('topic name'),
+        );
+        await tester.tap(topicFinder);
+        await tester.pumpAndSettle();
+
+        check(pushedRoutes).single
+          .isA<WidgetRoute>()
+          .page.isA<MessageListPage>()
+          .initNarrow.equals(TopicNarrow(channel.streamId, 'topic name'));
+      });
+
+      testWidgets('does not navigate on tapping topic in TopicNarrow', (tester) async {
+        final pushedRoutes = <Route<void>>[];
+        final navObserver = TestNavigatorObserver()
+          ..onPushed = (route, prevRoute) => pushedRoutes.add(route);
+
+        final channel = eg.stream();
+        final message = eg.streamMessage(stream: channel, topic: 'topic name');
+
+        await setupMessageListPage(
+          tester,
+          narrow: TopicNarrow(channel.streamId, 'topic name'),
+          navObservers: [navObserver],
+          streams: [channel],
+          messages: [message],
```

Comment:
nit: also put these in a *consistent* order between these related test cases, to make it easier for a reader to compare